book/text.md

```diff
@@ -298,35 +298,47 @@ a time:[^whitespace]
     including for identifying word boundaries.
 
 ``` {.python expected=False}
-w = font.measure(word)
-if cursor_x + w > WIDTH - HSTEP:
-    cursor_y += font.metrics("linespace") * 1.25
-    cursor_x = HSTEP
-self.display_list.append((cursor_x, cursor_y, word))
-cursor_x += w + font.measure(" ")
-```
-
-There's a lot of moving parts to this code. First, we measure the
-width of the text, and store it in `w`. We'd normally draw the text at
-`cursor_x`, so its right end would be at `cursor_x + w`, so we check
-if that's past the edge of the page. If it is, we make space by
-wrapping to the next line. Now we have the location to *start* drawing
-the word, so we add to the display list; and finally we update
-`cursor_x` to point to the end of the word.
-
-There are a few surprises in this code. One is that I call `metrics`
-with an argument; that just returns the named metric directly. Also, I
-increment `cursor_x` by `w + font.measure(" ")` instead of `w`. That's
-because I want to have spaces between the words: the call to `split()`
-removed all of the whitespace, and this adds it back. I don't add the
-space to `w` on the second line, though, because you don't need a
-space after the last word on a line.
+def layout(text):
+    # ...
+    for word in text.split():
+        # ...
+    return display_list
+```
+
+Unlike Chinese characters, words are different sizes, so we need to
+measure the width of each word:
+
+``` {.python expected=False }
+def layout(text):
+    font = tkinter.font.Font()
+    # ...
+    for word in text.split():
+        w = font.measure(word)
+    # ...
+```
+
+Here I've chosen to use Tk's default font. Now, if we draw the text at
+`cursor_x`, its right end would be at `cursor_x + w`. That might be
+past the right edge of the page, and in this case we need to make
+space by wrapping to the next line:
+
+``` {.python expected=False}
```

Comment:
Why expected=false?

Reply:
Similar to above, though at least in principle we could make this match _something_ by dropping the cursor_x/y lines (which move into `flush`) and at least matching the if statement.